### PR TITLE
Add panos_api_key module

### PR DESCRIPTION
**Pull request by freakinhippie**
Adding new module to retrieve the api_key for a given username/password on a device.

### CODE REVIEW

File: library/panos_api_key.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: panos_api_key
+short_description: retrieve api_key for username/password combination
+description:
+    - This module will allow retrieval of the api_key for a given username/password
+author: "Joshua Colson (@freakinhippie)"
+version_added: "2.8"
```
Comment (shinmog) [inline library/panos_api_key.py]: Should be 2.9

File: examples/fw_api_key.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  roles:
+    - role: PaloAltoNetworks.paloaltonetworks
+
+  tasks:
+  - name: Grab the credentials from ansible-vault
+    include_vars: 'firewall-secrets.yml'
+    no_log: 'yes'
+
+  - name: Retrieve API Key
+    panos_admin:
```
Comment (shinmog) [inline examples/fw_api_key.yml]: Shouldn't this be `panos_api_key`?

File: docs/modules/panos_api_key_module.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| --- | --- | --- | --- | --- |
+| ip_address | yes |  |  | IP address (or hostname) of PAN-OS device being configured. |
+| username |  | admin |  | Username credentials to use for auth unless *api_key* is set. |
+| password | yes |  |  | Password credentials to use for auth unless *api_key* is set. |
+
+## Examples
+
+    # Get the api_key of user admin
+      - name: retrieve admin api_key
+        panos_admin:
```
Comment (shinmog) [inline docs/modules/panos_api_key_module.md]: Shouldn't this be `panos_api_key`?

**Comment by freakinhippie** [comment]
Yes, thanks for the feedback.

**Pull request merged**